runtime/test/test_argument_conversion.cpp

```diff
@@ -15,40 +15,67 @@
 #include "common/ArgumentConversion.h"
 #include "cudaq/Optimizer/Dialect/CC/CCDialect.h"
 #include "cudaq/Optimizer/Dialect/Quake/QuakeDialect.h"
+#include "cudaq/qis/pauli_word.h"
 #include "mlir/InitAllDialects.h"
 #include "mlir/Parser/Parser.h"
+#include <sstream>
 
-void doSimpleTest(mlir::MLIRContext *ctx, const std::string &typeName,
-                  std::vector<void *> args) {
-  std::string code = R"#(
-func.func private @callee(%0: )#" +
-                     typeName + R"#()
-func.func @__nvqpp__mlirgen__testy(%0: )#" +
-                     typeName + R"#() {
-  call @callee(%0) : ()#" +
-                     typeName + R"#() -> ()
-  return
-})#";
+void doTest(mlir::MLIRContext *ctx, std::vector<std::string> &typeNames,
+            std::vector<void *> args, std::size_t startingArgIdx = 0) {
+
+  // Create code
```

Comment:
Why do we need a stringstream?

Reply:
What would be the preference?

Reply:
I switched to `llvm::raw_string_ostream` (similar to what we use in creating argument substitutions when calling argumnet conversion). Let me know if there is something better to use.

Reply:
Originally, it was just concatenating pieces of strings, right?